network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/end/BetaEndedDialogMonitor.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.networkprotection.impl.waitlist.end
+
+import android.app.Activity
+import com.duckduckgo.app.di.AppCoroutineScope
+import com.duckduckgo.browser.api.ActivityLifecycleCallbacks
+import com.duckduckgo.common.utils.DispatcherProvider
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.networkprotection.impl.waitlist.NetPRemoteFeature
+import com.duckduckgo.networkprotection.impl.waitlist.store.NetPWaitlistRepository
+import com.squareup.anvil.annotations.ContributesMultibinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+@ContributesMultibinding(AppScope::class)
+@SingleInstanceIn(AppScope::class)
+class BetaEndedDialogMonitor @Inject constructor(
```

Comment:
I'd move this into the `NetpVpnAccessRevokedDialogMonitor` to prioritise what dialog shows in case both conflict. Otherwise the UX might be confusing in such case